src/actions/PermissionsActions.js

```diff
@@ -0,0 +1,46 @@
+import {Location, Permissions} from 'expo';
+
+import {
+    LOCATION_SERVICES_PERMISSION_UPDATED,
+    CAMERA_PERMISSION_UPDATED,
+    CAMERA_ROLL_PERMISSION_UPDATED,
+    CURRENT_LOCATION_UPDATED
+} from './types';
+
+import {permissionResponses} from '../constants';
+
+const {DENIED} = permissionResponses;
+
+export const getCurrentUserLocation = () => {
+    return async (dispatch) => {
+        try {
+            const location = await Location.getCurrentPositionAsync({enableHighAccuracy: true});
+            const {coords: {longitude, latitude}} = location;
+
+            return dispatch({
+                type: CURRENT_LOCATION_UPDATED,
+                payload: {currentLocation: {longitude, latitude}}
+            })
+
+        } catch (error) {
+            return dispatch({
+                type: LOCATION_SERVICES_PERMISSION_UPDATED,
+                payload: {locationServicesPermission: DENIED}
+            })
+        }
```

Comment:
Above, had to use a try/catch in case the user went into settings and changed from 'allow' to 'deny' for location services (we'd have no way of knowing that they did that, so we'd try to grab their location and get an error).

Reply:
wow, really cool!